### ISSUE TITLE
Interactive Design Meeting Notes - 7/17/15 & 7/24/15

### ISSUE THREAD

**Issue by kuhlenh** [dotnet/roslyn#4110]
# Interactive Design Meeting 7/8/15

Design Team
-------------
The following individuals are members of the dream team:
[Andrew Casey](https://github.com/amcasey)
[Anthony D. Green](https://github.com/AnthonyDGreen)
[Chuck Stoner](https://github.com/CharlesStoner)
[Dustin Campbell](https://github.com/DustinCampbell)
[Kasey Uhlenhuth](https://github.com/kuhlenh)
[Kevin Halverson](https://github.com/KevinH-MS)
[Kevin Pilch-Bisson](https://github.com/Pilchie)
[Lincoln Atkinson](https://github.com/latkin)
[Manish Jayaswal](https://github.com/ManishJayaswal)
[Matt Warren](https://github.com/mattwar)
[Tomas Matousek](https://github.com/tmat)

------------------

AGENDA - Identify Core Experiences
-------------------
Below are a list of the core experiences for the Interactive Window and scripting we want to support by 1.1. The brackets contain the status of each work item. 

### Interactive APIs for 1.1
- Hosting API extensibility point so host can load .nupkg and script **[IMPLEMENT & TEST]**
- Submission inherit references **[IMPLEMENT & TEST]**
- Cross-platform execution of scripts **[IMPLEMENT & TEST...CONTINGENT ON CORE]**
- Other rando bugs **[FIX & TEST]**

### Core Interactive Window features for 1.1
- Full C# language support **[TEST & BUG FIX]**
- Core C# editing experience supported for Interactive Window **[INVESTIGATE & TEST & BUG FIX]**
- #r .dlls and nupkgs **[.dll: TEST & BUG FIX // .nupkg: IMPLEMENT & TEST]**
- History of commands **[POLISH & TEST]**
- Help/documentation commands in Interactive Window (#help) **[POLISH & TEST]**
- Shadowing in the Interactive Window **[TEST & BUG FIX]**
- Copy REPL session without prompts **[IMPLEMENT & TEST]**
- Telemetry **[IMPLEMENT & TEST]**
- #load scripts **[IMPLEMENT & TEST]**
	
### Core Script features for 1.1
- Full C# language support **[TEST & BUG FIX]**
- Core C# editing experience supported for scripts **[INVESTIGATE & TEST & BUG FIX]**
- #r .dlls and nupkgs **[.dll: TEST & BUG FIX // .nupkg: IMPLEMENT & TEST]**
- Provide a way to run script outside VS **[COORDINATE W SCRIPTCS]**
- Provide a way to run script within VS **[DESIGN & IMPLEMENT]**
- C# scripts in File New **[IMPLEMENT & TEST]**
- Telemetry **[IMPLEMENT & TEST]**
- #load scripts **[IMPLEMENT & TEST]**

### Stretch features for 1.1
- Completion for #r with installed NuGets **[IMPLEMENT & TEST]**
- Debug scripts **[DESIGN & IMPLEMENT]**
- Send snippet from editor to Interactive Window **[TEST & BUG FIX]**

### Non-goals for 1.1
- REPL attached to debugger process
- Debugging inside of REPL session (no BPs or stepping inside of Interactive Window)

**Comment by paul1956** [comment]
Is there an equivalent meeting for VB? If so do I need to do something special to subscribe to it?

**Comment by RichiCoder1** [comment]
Usual blurb where I mention that Cake (https://cake-build.net) is also very interested in the progress of these features. Awesome work guys!

**Comment by dfkeenan** [comment]
I am looking forward to being able to host scripting in my own project(s).  Hoping some of the stuff in the Features APIs that @DustinCampbell is working on will mean I will be able to give a good editing experience outside of VS as well.

Will script hosting work on the UWP?

**Comment by dfkeenan** [comment]
Something else I have been thinking about from a hosting/plugin system is being able to make the script meet some contract i.e. implement an interface or inherit from a base class.  Although being able to verify that the script includes a class that meets the requirement (this already exists) and being able to get an instance of it would be just as good.

**Comment by tmat** [comment]
@dfkeenan No, scripting won't work in UWP since runtime code gen is not supported. 

Re interface - we have exactly that work item on our backlog.

**Comment by dfkeenan** [comment]
@tmat I suppose it's the same story for Windows store apps?  Though I think I read somewhere that MS is changing store policy to allow full dotnet framework/win32 apps in the store for desktop.

Re interface - :smile:

**Comment by tmat** [comment]
@dfkeenan The Store and the platform are two different things. The UWP platform doesn't support runtime code gen. You'll be able to run scripts in your Desktop app, which you'll be able to add to the Store if the Store policy changes as you say.

**Comment by MgSam** [comment]
IMO these are all worthwhile features goals but way more than you need for an initial release. Even having the REPL alone without any of the scripting stuff would provide tremendous value. I vote to get something in the hands of users sooner rather than waiting until everything and the kitchen sink has been built.

**Comment by ManishJayaswal** [comment]
@paul1956  We are following the usual Roslyn framing model of starting with C#. We are doing VB work simultaneously where it makes sense, from efficiency point of view. VB previews and releases will follow C#.

**Comment by Kemaletdin** [comment]
I am interested in playing with the Scripting API that should be in 1.1 so I looked at the MyGet it has v2.0 beta 1 and 1.1 is not released so are you skipping 1.1 and going for v2.0 if not where do I get the nightly for v1.1.  thanks

**Comment by ManishJayaswal** [comment]
@Kemaletdin the v2.0 beta 1 packages in MyGet are the correct nightly packages. We just haven't fixed the versioning of scripting packages to be different from rest of the Roslyn packages yet.

**Comment by aluanhaddad** [comment]
I want to second what @MgSam said.
